packages/hardhat-core/src/builtin-tasks/compile.ts

```diff
@@ -684,6 +685,14 @@ subtask(TASK_COMPILE_SOLIDITY_RUN_SOLC)
       solcPath: string;
       solcVersion?: string;
     }) => {
+      if (solcVersion !== undefined && semver.parse(solcVersion) === null) {
```

Comment:
Sorry, I should've commented this before too, but it's a minor readability change:

```suggestion
      if (solcVersion !== undefined && semver.valid(solcVersion)) {
```